protobuf/src/coded_input_stream.rs

```diff
@@ -605,10 +605,10 @@ impl<'a> CodedInputStream<'a> {
     }
 
     /// Read message, do not check if message is initialized
-    pub fn merge_message<M: Message>(&mut self, message: &mut M) -> ProtobufResult<()> {
+    pub fn merge_message(&mut self, message: &mut dyn MessageDyn) -> ProtobufResult<()> {
```

Comment:
This change may have negative performance implications (extra virtual call). Maybe that's fine, maybe not, but changes like that should not quietly slip.

Reply:
Alternatively, is it save to downcast manually here?
https://github.com/stepancheg/rust-protobuf/blob/55eba4b39ff54cf19de03558a0e30ffc67ecf39b/protobuf/src/reflect/dynamic/mod.rs#L445

Reply:
No it's not, because `msg_inst` can be a generated message, not dynamic: dynamic message can depend on generated message (depending on `google.protobuf.DescriptorProto` is the easiest example).

I think for now we can have a neighbour function like `merge_message_dyn`.